### PR TITLE
MC7000 - describe changes for Mixxx 2.3.1

### PR DESCRIPTION
**Pull request by toszlanyi**
correlates to https://github.com/mixxxdj/mixxx/pull/4021

### CODE REVIEW

File: source/hardware/controllers/denon_mc7000.rst [hunk 1]
```diff
@@ -152,7 +152,9 @@ Library Section
    "51", ":hwlabel:`L.PREP`",                         "Press to load the currently selected track to the Preview Deck."
    "51", ":hwlabel:`SHIFT` + :hwlabel:`L.PREP`",      "Press to start and stop playing the track in Preview Deck."
    "51", ":hwlabel:`SORT` + :hwlabel:`L.PREP`",       "Press to sort the tracks by their Key."
-   "52", ":hwlabel:`FILES`",                          "Press to maximise the library. Press this button again to exit maximised library."
+   "52", ":hwlabel:`FILES`",                          "Press to maximise the library. Press this button again to exit maximised library.
```
Comment (ronso0) [inline source/hardware/controllers/denon_mc7000.rst:155]:
```suggestion
   "52", ":hwlabel:`FILES`",                          "Press to maximize the library. Press this button again to exit maximized library.
```
z for consistency ; )
here and below

File: source/hardware/controllers/denon_mc7000.rst [hunk 2]
```diff
@@ -181,7 +183,9 @@ Deck Section
 
    When the :hwlabel:`VINYL` button is lit, move the platter to 'scratch' the track as you would with a vinyl record.
 
-   When the :hwlabel:`VINYL` button is unlit (or if you are touching only the side of the platter), move the platter to temporarily adjust the track’s speed."
+   When the :hwlabel:`VINYL` button is unlit (or if you are touching only the side of the platter), move the platter to temporarily adjust the track’s speed.
+
+   If the library is maximized (use :hwlabel:`FILES` button) then turning the jog wheel will quickly navigate through the actual playlist or crate. No interaction is done with the current playing track. Be careful though, after exit the maximized library view the jog wheel will immediately return to jog or scratch mode."
```
Comment (ronso0) [inline source/hardware/controllers/denon_mc7000.rst:188]:
```suggestion
   If the library is maximized (use :hwlabel:`FILES` button) turning the jog wheel will scroll the current table view (currently playing track is not affected). Be careful though: after exiting the maximized library view the jog wheel will immediately return to jog or scratch mode."
```

File: source/hardware/controllers/denon_mc7000.rst [hunk 3]
```diff
@@ -219,6 +223,9 @@ Deck Section
 .. hint::
    The jog sensitivity and scratch parameters (28), the default vinyl mode (30), the pitch fader ranges (32) as well as the needle drop activity (54) can be set by modifying the variables inside the :ref:`JavaScript file<denon_mc7000_uservariables>` accordingly.
 
+.. note::
+   Use :hwlabel:`SHIFT` + :hwlabel:`CENSOR` or :hwlabel:`SHIFT` + jog wheel side for a backspin while :hwlabel:`SLIP` is active. Backspin using the platter is only possible while :hwlabel:`SLIP` is disabled.
```
Comment (ronso0) [inline source/hardware/controllers/denon_mc7000.rst:227]: press Shift + Censor _OR_ press Shift + turn wheel?
If so this needs to be clarified IMO

**Review by ronso0 — CHANGES_REQUESTED** [review]

File: source/hardware/controllers/denon_mc7000.rst (hunk 3 above)
Comment (toszlanyi) [reply source/hardware/controllers/denon_mc7000.rst:227]: Shift + jog wheel is a side effect from seeking through a track. Maybe it's better I just delete the comment and only mention the dedicated backspin function on the censor button. Gonna send an update later today. Cheers!

File: source/hardware/controllers/denon_mc7000.rst (hunk 2 above)
Comment (toszlanyi) [reply source/hardware/controllers/denon_mc7000.rst:188]: done

File: source/hardware/controllers/denon_mc7000.rst (hunk 1 above)
Comment (toszlanyi) [reply source/hardware/controllers/denon_mc7000.rst:155]: done

File: source/hardware/controllers/denon_mc7000.rst [hunk 4]
```diff
@@ -201,11 +205,15 @@ Deck Section
...
-   "44", ":hwlabel:`SLIP`",                                "Press to enable or disable Slip Mode. In Slip Mode, you can jump to cue points, trigger loops or use the platters, while the track’s timeline continues. In other words, when you deactivate Slip Mode, the track will resume normal playback from where it would have been if you had never done anything (i.e., as if the track had been playing forward the whole time)."
+   "44", ":hwlabel:`SLIP`",                                "Press to enable or disable Slip Mode. In Slip Mode, you can jump to cue points, trigger loops or use the platters, while the track’s timeline continues. In other words, when you deactivate Slip Mode, the track will resume normal playback from where it would have been if you had never done anything (i.e., as if the track had been playing forward the whole time).
+
+   After scratch, hot cue play, reverse play and backspin the slip mode releases itself so the timeline of the track jumps back to original position after the action is done. Slip mode reactivates itself after 50ms so you can continue with more actions in slip mode almost immediately after."
```
Comment (ronso0) [inline source/hardware/controllers/denon_mc7000.rst:210]: This sounds like an interesting feature!

File: source/hardware/controllers/denon_mc7000.rst [hunk 5]
```diff
@@ -316,6 +329,9 @@ Use :hwlabel:`SHIFT` + pad button to stop a sampler while playing or eject a sam
    :alt: Denon MC7000 SAMPLER Mode
    :figclass: pretty-figures
 
+.. hint::
+   To multi play triggered samplers at the same time you can set the user variable ``MC7000.prevSamplerStop`` inside the :ref:`JavaScript file<denon_mc7000_uservariables>` to ``false``.
```
Comment (ronso0) [inline source/hardware/controllers/denon_mc7000.rst:333]:
```suggestion
   To use default Mixxx behaviour and allow playing multiple samplers at the same time you can set the user variable ``MC7000.prevSamplerStop`` inside the :ref:`JavaScript file<denon_mc7000_uservariables>` to ``false``.
```

**Review by ronso0 — CHANGES_REQUESTED** [review]

File: source/hardware/controllers/denon_mc7000.rst (hunk 5 above)
Comment (ronso0) [reply source/hardware/controllers/denon_mc7000.rst:333]: That's it from me. LGTM when the line above is changed!
Someone who reviewed the actual mapping should take a look though.

Comment (toszlanyi) [reply source/hardware/controllers/denon_mc7000.rst]: done

File: source/hardware/controllers/denon_mc7000.rst (hunk 4 above)
Comment (toszlanyi) [reply source/hardware/controllers/denon_mc7000.rst:210]: Yes it is. There was a discussion on [zulip chat](https://mixxx.zulipchat.com/#narrow/stream/109122-general/topic/Slip.20Mode/near/240571349) that triggered me to add it to the MC7000 mapping. Short timer after the release to enable SLIP again. 50ms seems to be very good. Short enough you could not press a button (incl platter) twice that quickly. Long enough not to miss something. The 50ms is a pure empiric value though

**Comment by Be-ing** [comment]
@ronso0 please merge if this is ready and let's get 2.3.1 released.

**Comment by ronso0** [comment]
oh, for some reason I did not get notified about your post.
LGTM, thank you!

**Pull request merged**